[WIP] Adding Embedded Wizard

cshinn opened the pull request:
## Description
A first pass at adding documentation for embedded wizards

[Tracked Here](https://patternfly.atlassian.net/browse/OSUX-107)

## Use Cases
- User must fill out the wizard in order to take any further action in the application
- Filling out the wizard will conclude the user's interaction with the website or application

## Requirements
- [x] Must allow for the same features and functions of a modal wizard
- [x] Must fit within standard PatternFly page layouts
- [x] Must allow navigation when necessary

## Changes
- Adds image of an embedded wizard to the Overview page
- Adds a section for embedded wizards in the Design page as well as guidelines for when to use each type of wizard

Mgranfie commented:
@cshinn this looks good! Not sure if alternate convo's took place and you are all set, as I have a few small pieces of feedback that are really nits. 
1. I see you removed the X button, and that was the first thing I looked for, as a physical indicator and feedback that this page is embedded.
2. The Wizard title feels tight and forced in the space, was there any wiggle room to provide more white space there?
3. The buttons moved from right to left, curious your thinking there. They felt more natural on the right, as this is a fairly common standard and indicates progress forward, as far as order of events.

None of these suggestions are critical path, so if this is wrapped up... np. Just didn't want to not acknowledge all your work!

beanh66 commented:
@cshinn This looks good! Just one thing I notice is at the bottom of this pattern there is a section called "What’s not covered in the current design but will be covered in future sprints:" and #5 talks about embedded wizards! I think it's safe to say you can remove that bullet 👍

mcarrano commented on pattern-library/communication/wizard/design/design.md:
```diff
@@ -38,6 +38,24 @@ In the example above, the back button is disabled since this shows the first sub
+![Embedded wizard](img/embedded-wizard-callout.png)
+
+**Differences from Modal Wizards** Embedded wizards follow the same guidelines as modal wizards, with a few exceptions.
+1. **Button Bar:** In an embedded wizard, the action buttons are placed in the bottom left corner as they would be in a [Full Page Form](http://www.patternfly.org/pattern-library/forms-and-controls/forms/#/within-a-page).
```
The Forms pattern you are referencing in deprecated and planned for removal. Please refer to the Buttons on Forms pattern for updated content: http://www.patternfly.org/pattern-library/forms-and-controls/buttons-on-forms/

mcarrano replied:
Also, are you calling for a fixed footer here?  I think you are, in which case you should probably call that out.

cshinn commented:
@Mgranfie Do you mean that the embedded version should have the X Button, or that taking it out was the right move?

@beanh66 HA, good catch!

@mcarrano I'll make those changes. re: the number bar,  it's made me think about how this would work on a mobile device. An important detail if it needs to scale with the whole page. I will update this with suggestions

cshinn commented:
@mcarrano @Mgranfie I've updated the PR with your suggestions. I've fixed the spacing around the title, updated the link to the buttons page, and added a bit about the footer. I'll be working on a design for a responsive wizard in the next couple days, and will address #347 in a future PR

mcarrano commented:
I have approved this PR. Thanks for making the updates @cshinn .  @Mgranfie can you take a look, also? Once you approve I will go ahead and merge this.

Mgranfie commented:
LGTM!
